ynr/apps/elections/api/next/api_views.py

```diff
@@ -0,0 +1,164 @@
+from candidates import models as extra_models
+from candidates.api.next.serializers import LoggedActionSerializer
+from elections.models import Election
+from elections.uk.geo_helpers import (
+    get_ballots_from_coords,
+    get_ballots_from_postcode,
+)
+from elections.filters import BallotFilter
+from official_documents.models import OfficialDocument
+from popolo.models import Membership
+from utils.db import LastWord
+
+
+class UpcomingElectionsView(View):
+
+    http_method_names = ["get"]
+
+    def get(self, request, *args, **kwargs):
+        postcode = request.GET.get("postcode", None)
+        coords = request.GET.get("coords", None)
+
+        # TODO: postcode may not make sense everywhere
+        errors = None
+        if not postcode and not coords:
+            errors = {"error": "Postcode or Co-ordinates required"}
+
+        try:
+            if coords:
+                ballots = get_ballots_from_coords(coords)
+            else:
+                ballots = get_ballots_from_postcode(postcode)
+        except Exception as e:
+            errors = {"error": e.message}
+
+        if errors:
+            return HttpResponse(
+                json.dumps(errors), status=400, content_type="application/json"
+            )
+
+        results = []
+        ballots = ballots.select_related("post", "election")
+        for ballot in ballots:
+            results.append(
+                {
+                    "post_name": ballot.post.label,
+                    "post_slug": ballot.post.slug,
+                    "organization": ballot.post.organization.name,
+                    "election_date": text_type(ballot.election.election_date),
+                    "election_name": ballot.election.name,
+                    "election_id": ballot.election.slug,
+                }
+            )
+
+        return HttpResponse(
+            json.dumps(results), content_type="application/json"
+        )
+
+
+class ElectionViewSet(viewsets.ReadOnlyModelViewSet):
+    lookup_value_regex = r"(?!\.json$)[^/]+"
+    queryset = Election.objects.order_by("id")
+    lookup_field = "slug"
+    serializer_class = elections.api.next.serializers.ElectionSerializer
+    filterset_fields = ("current",)
+    pagination_class = ResultsSetPagination
+
+
+class BallotViewSet(viewsets.ReadOnlyModelViewSet):
+    """
+    A paginated list of all ballots
+
+    """
+
+    lookup_field = "ballot_paper_id"
+    lookup_value_regex = "[^/]+"
+    queryset = (
+        extra_models.Ballot.objects.select_related("election", "post")
+        .prefetch_related(
+            Prefetch(
+                "membership_set",
+                queryset=Membership.objects.all()
+                .select_related("result", "person", "party")
+                .annotate(last_name=LastWord("person__name"))
+                .order_by(
+                    "-elected",
+                    "-result__is_winner",
+                    "-result__num_ballots",
+                    "person__sort_name",
```

Comment:
Would it make sense to sort by

```
...
"-result__num_ballots",
"party_list_position",
"person__sort_name",
...
```

here so that `party_list_position` takes precedence over name if we have no results?